### ISSUE TITLE
Duplicate literals in query cause egglog to panic

### ISSUE THREAD

**Issue by yihozhang** [egraphs-good/egglog#133]
The following program panics.

```
(relation f (i64))

(rule ((= x 1)
       (= x 2))
      ((f x)))
      
(run 1)
(print f)
```

**Comment by mwillsey** [comment]
What’s the desired behavior here? A warning? Type checking error?

**Comment by yihozhang** [comment]
I think semantically this should just be a valid although unfirable rule. It should have the same semantics as the following rule (which also panics currently)
```
(relation f (i64))

(rule ((= 2 1))
      ((f 1)))
      
(run 1)
(print f)
```

A warning would be nice but not necessarily essential.

**Comment by yihozhang** [comment]
Fixed by #219
